x-pack/plugins/apm/public/components/app/metrics/index.tsx

```diff
@@ -31,5 +33,20 @@ export function Metrics() {
     return <ServerlessMetrics />;
   }
 
+  const dashboardFile = hasDashboardFile({
+    agentName,
+    runtimeName,
+    serverlessType,
+  });
```

Comment:
Is an actual file required? How will end users be able to add their own custom visualisations?

The way I expected this to work, is that we allow end users to create custom visualisations in Lens, and then by cross-referencing them in APM they can choose to embed them. Visualisations should probably receive `service.name` and `service.environment` as params.

Reply:
Resolved after talking offline